pkg/bootstrap/docker/dockermachine/helper.go

```diff
@@ -147,10 +150,32 @@ func Client(name string) (*docker.Client, error) {
 		client, err = docker.NewVersionedClient(dockerHost, "")
 	}
 	if err != nil {
-		return nil, errors.NewError("could not get Docker client for machine %s", name).WithCause(err)
+		return nil, nil, errors.NewError("could not get Docker client for machine %s", name).WithCause(err)
 	}
 	client.SkipServerVersionCheck = true
-	return client, nil
+
+	var httpClient *http.Client
+	if len(certPath) > 0 {
```

Comment:
they're all assumed to exist by docker code (and are required as far as I can tell):
https://github.com/docker/engine-api/blob/master/client/client.go#L40

The code here is doing the same thing that their client is doing. The only reason this code is here is that they don't allow passing in the values of env vars, and get the vars themselves.